src/Illuminate/Support/Str.php

```diff
@@ -1778,6 +1778,24 @@ public static function wordWrap($string, $characters = 75, $break = "\n", $cutLo
         return wordwrap($string, $characters, $break, $cutLongWords);
     }
 
+    /**
+     * Counts the number of lines in the input string.
+     *
+     * @param  string  $string
+     * @param  bool  $withEmptyLines
+     * @return int
+     */
+    public static function lineCount(string $string, bool $withEmptyLines = true): int
+    {
+        $string = str_replace("\r\n", "\n", $string);
+
+        if ($withEmptyLines) {
+            return count(explode("\n", $string));
+        }
```

Comment:
You can avoid replacing the newline variations first, by doing
```suggestion
        if ($withEmptyLines) {
            return preg_match_all('/\r\n|\r|\n/', $string));
        }
```